api/jobs_test.go

```diff
@@ -57,27 +60,40 @@ func TestCreateJobHandler(t *testing.T) {
 		},
 	}
 
+	indexerMock := &apiMock.IndexerMock{
+		CreateIndexFunc: func(ctx context.Context, serviceAuthToken, searchAPISearchURL string, httpClient dpHTTP.Clienter) (*http.Response, error) {
+			resp := &http.Response{
+				StatusCode: 201,
+			}
+			return resp, nil
+		},
+		GetIndexNameFromResponseFunc: func(ctx context.Context, body io.ReadCloser) (string, error) {
+			return "ons1638363874110115", nil
+		},
+	}
+
 	Convey("Given a Search Reindex Job API that can create valid search reindex jobs and store their details in a Data Store", t, func() {
 		api.NewID = func() string { return validJobID1 }
 		cfg, err := config.Get()
 		So(err, ShouldBeNil)
-		apiInstance := api.Setup(ctx, mux.NewRouter(), dataStorerMock, &apiMock.AuthHandlerMock{}, taskNames, cfg)
-		createJobHandler := apiInstance.CreateJobHandler(ctx)
+		httpClient := dpHTTP.NewClient()
+
+		apiInstance := api.Setup(mux.NewRouter(), dataStorerMock, &apiMock.AuthHandlerMock{}, taskNames, cfg, httpClient, indexerMock)
 
 		Convey("When a new reindex job is created and stored", func() {
 			req := httptest.NewRequest("POST", "http://localhost:25700/jobs", nil)
 			resp := httptest.NewRecorder()
 
-			createJobHandler.ServeHTTP(resp, req)
+			apiInstance.CreateJobHandler(resp, req)
 
 			Convey("Then the newly created search reindex job is returned with status code 201", func() {
 				So(resp.Code, ShouldEqual, http.StatusCreated)
-				payload, err := ioutil.ReadAll(resp.Body)
+				payload, err := io.ReadAll(resp.Body)
 				So(err, ShouldBeNil)
 				newJob := models.Job{}
 				err = json.Unmarshal(payload, &newJob)
 				So(err, ShouldBeNil)
-				expectedJob, err := models.NewJob(validJobID1)
+				expectedJob, err := ExpectedJob(validJobID1, zeroTime, 0, zeroTime, zeroTime, zeroTime, "ons1638363874110115", "created", 0, 0)
```

Comment:
I'm not an expert on Go conventions but this function with loads of arguments would be undesirable in other languages. Can't we just do `expectedJob, err := models.NewJob(validJobID1)` as done elsewhere? If the test is that we want to check all fields are return the same then we would need to assign them to something other than the defaults wouldn't we?

Reply:
Somebody else thinks this is the right way (?)